src/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -127,48 +83,30 @@ public static FileStream Create(string path)
         // The file is opened with ReadWrite access and cannot be opened by another 
         // application until it has been closed.  An IOException is thrown if the 
         // directory specified doesn't exist.
-        //
-        // Your application must have Create, Read, and Write permissions to
-        // the file.
-        // 
         public static FileStream Create(string path, int bufferSize)
-        {
-            return new FileStream(path, FileMode.Create, FileAccess.ReadWrite, FileShare.None, bufferSize);
-        }
+            => new FileStream(path, FileMode.Create, FileAccess.ReadWrite, FileShare.None, bufferSize);
 
         public static FileStream Create(string path, int bufferSize, FileOptions options)
-        {
-            return new FileStream(path, FileMode.Create, FileAccess.ReadWrite,
-                                  FileShare.None, bufferSize, options);
-        }
+            => new FileStream(path, FileMode.Create, FileAccess.ReadWrite, FileShare.None, bufferSize, options);
  
         // Deletes a file. The file specified by the designated path is deleted.
         // If the file does not exist, Delete succeeds without throwing
         // an exception.
         // 
         // On NT, Delete will fail for a file that is open for normal I/O
-        // or a file that is memory mapped.  
-        // 
-        // Your application must have Delete permission to the target file.
-        // 
+        // or a file that is memory mapped.
         public static void Delete(string path)
         {
             if (path == null)
                 throw new ArgumentNullException(nameof(path));
 
-            string fullPath = Path.GetFullPath(path);
-
-            FileSystem.DeleteFile(fullPath);
+            FileSystem.DeleteFile(Path.GetFullPath(path));
```

Comment:
Do we always call GetFullPath immediately on paths passed in to IO API? This is presumably in case current directory changes?

Reply:
Yes, we always call GetFullPath().  The current directory thing would impact things, and in some cases we need to know that the path is normalized. Historically I'm sure it was more about CAS and the preemptive checks we made in GFP than anything else.